elixir-mode.el

```diff
@@ -107,6 +126,11 @@
                       (zero-or-more (any "a-z" "A-Z" "0-9" "_" "\"" "'" "!" "@" "?")))
                      (and "\"" (one-or-more (not (any "\""))) "\"")
                      (and "'" (one-or-more (not (any "'"))) "'"))))
+      (numbers . ,(rx (and symbol-start
+                           (? "-")
+                           (+ digit)
+                           (0+ (and "_" (= 3 digit)))
+                           symbol-end)))
```

Comment:
I think this is too naive for parsing numbers in Elixir.  I propose the following:

```suggestion
      (numbers . ,(rx (and symbol-start
                               (zero-or-one "-")
                               (one-or-more digit)
                               (zero-or-more (or digit (seq ?_ digit)))
                               (zero-or-one ".")
                               (zero-or-more (or digit (seq ?_ digit)))
                               symbol-end)))
```

This will match things like:

```elixir
0
-0
0_0
-0_0.0_0
```

I know they are not what the formatter will typically do but I see no reason why we should bind this face to only what the formatter does.

Not sure how this will behave inside comments, heredocs and so on though.

Also I am trying to follow the use of `zero-or-more` and friends because of what the rest of the file is using. 

Tests would be good here too if possible.

Reply:
If you prefer that, that's fine. I'll leave it your capable hands to get this across the finish line 🙏